sklearn/calibration.py

```diff
@@ -188,7 +190,21 @@ def fit(self, X, y, sample_weight=None):
                                               sample_weight[test])
                 else:
                     calibrated_classifier.fit(X[test], y[test])
-                self.calibrated_classifiers_.append(calibrated_classifier)
+                    if hasattr(calibrated_classifier.base_estimator, 'oob_decision_function_'):
+                        idx_pos_class = calibrated_classifier.label_encoder_.transform(
+                            calibrated_classifier.base_estimator.classes_)
+                        df = calibrated_classifier.base_estimator.oob_decision_function_[:, 1:]
+                        oob[train, :] = calibrated_classifier.transform_proba(
+                            df, idx_pos_class, len(self.classes_),
+                            calibrated_classifier.base_estimator.oob_decision_function_)
+                        oob_decision_function_.append(oob)
+                        # to save memory we can remove the oob from the base estimator
+                        delattr(calibrated_classifier.base_estimator, 'oob_decision_function_')
+                        calibrated_classifier.base_estimator.oob_score = False
+                    self.calibrated_classifiers_.append(calibrated_classifier)
+                if hasattr(self.base_estimator, 'oob_score') and self.base_estimator.oob_score:
+                    self.oob_decision_function_ = np.nanmean(np.array(oob_decision_function_), axis=0)
+                    self.oob = True
```

Comment:
This seems redundant

Reply:
Moved this to the init to be consistent with model behavior.
This way if you initialize with a base learner that has`oob=True` then the calibrated classifier will also be initialized the same way.
The decision function, like in random forests, is only initialized when calling `fit`